src/components/ChannelPreview.js

```diff
@@ -1,116 +1,110 @@
+        setUnread(unread);
+      } else {
+        setUnread(0);
+      }
     };
-  }
 
-  static propTypes = {
-    /** **Available from [chat context](https://getstream.github.io/stream-chat-react/#chat)** */
-    channel: PropTypes.object.isRequired,
-    /** Current selected channel object */
-    activeChannel: PropTypes.object.isRequired,
-    /** Setter for selected channel */
-    setActiveChannel: PropTypes.func.isRequired,
-    /**
-     * Available built-in options (also accepts the same props as):
-     *
-     * 1. [ChannelPreviewCompact](https://getstream.github.io/stream-chat-react/#ChannelPreviewCompact) (default)
-     * 2. [ChannelPreviewLastMessage](https://getstream.github.io/stream-chat-react/#ChannelPreviewLastMessage)
-     * 3. [ChannelPreviewMessanger](https://getstream.github.io/stream-chat-react/#ChannelPreviewMessanger)
-     *
-     * The Preview to use, defaults to ChannelPreviewLastMessage
-     * */
-    Preview: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
-    /**
-     * Object containing watcher parameters
-     * @see See [Pagination documentation](https://getstream.io/chat/docs/#channel_pagination) for a list of available fields for sort.
-     * */
-    watchers: PropTypes.object,
-  };
-
-  static defaultProps = {
-    Preview: ChannelPreviewCountOnly,
-  };
+    channel.on('message.new', handleEvent);
+    return () => {
+      channel.off('message.new', handleEvent);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [channel]);
 
-  componentDidMount() {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => {
     // listen to change...
-    const channel = this.props.channel;
-    const unread = channel.countUnread();
-
-    this.setState({ unread });
-    channel.on('message.new', this.handleEvent);
-  }
-
-  componentWillUnmount() {
-    const channel = this.props.channel;
-    channel.off('message.new', this.handleEvent);
-  }
-
-  handleEvent = (event) => {
-    const channel = this.props.channel;
-    const isActive = this.props.activeChannel.cid === channel.cid;
-    if (!isActive) {
-      const unread = channel.countUnread(this.state.lastRead);
-      this.setState({ lastMessage: event.message, unread });
-    } else {
-      this.setState({ lastMessage: event.message, unread: 0 });
+    if (isActive()) {
+      setLastRead(new Date());
     }
-  };
 
-  componentDidUpdate(prevProps) {
-    if (this.props.activeChannel.cid !== prevProps.activeChannel.cid) {
-      const isActive = this.props.activeChannel.cid === this.props.channel.cid;
-      if (isActive) {
-        this.setState({ unread: 0, lastRead: new Date() });
-      }
-    }
-  }
+    const unread = channel.countUnread(lastRead);
+    setLatestMessage(getPreviewLatestMessage(channel));
+    setUnread(unread);
+  }, [channel, activeChannel]);
+
+  const isActive = useCallback(() => activeChannel.cid === channel.cid, [
+    activeChannel,
+    channel,
+  ]);
```

Comment:
not that it's a big deal but you can also leverage `useMemo` for something like this too which let you skip one function call
`const isActive = useMemo(() => activeChannel.cid === channel.cid, [activeChannel, channel]);`

Reply:
also since this function is pretty simple evaluation you can ditch both useCallback and useMemo and just evaluate it on every render, simple as: `const isActive = activeChannel.cid === channel.cid`